### PR TITLE
Peer

### PR DESCRIPTION
**Pull request by bbrietzke**
Need to be able to find the peer ( remote ) address of where the requests are coming from.

### CODE REVIEW

**Comment by josevalim** [comment]
Tests please? :) We also need a way to set the peer for test connections.

**Comment by bbrietzke** [comment]
Thanks for making me do the tests.  I learnt a few things!

**Comment by bbrietzke** [comment]
I added the tests and updated some of the documentation around the additional parameter.  If there is a better way or a more appropriate place to provide the test peer please let me know.

File: lib/dynamo/connection/test.ex [hunk 1]
```diff
@@ -10,31 +10,31 @@ defmodule Dynamo.Connection.Test do
...
   use Dynamo.Connection.Behaviour,
     [ :query_string, :raw_req_headers, :raw_req_body, :raw_req_cookies, :fetched,
-      :path, :path_segments, :sent_body, :original_method, :scheme, :port ]
+      :path, :path_segments, :sent_body, :original_method, :scheme, :port, :peer ]
 
   @doc """
   Initializes a connection to be used in tests.
   """
-  def new(method, path, body // "") do
+  def new(method, path, body // "", peer // {127, 0, 0, 1}) do
```
Comment (josevalim) [inline lib/dynamo/connection/test.ex]: The peer is not strictly required, so we don't need to pass it as parameter here. We just need to provide a `conn.peer({ 127, 0, 0, 1})` that allows people to set it if they want to. :)

File: lib/dynamo/connection/test.ex [hunk 2]
```diff
@@ -205,7 +210,7 @@ defmodule Dynamo.Connection.Test do
   This can be considered the counter-part of recycle
   (which is used to clean up the response).
   """
-  def req(method, path, body // "", conn) do
+  def req(method, path, body // "", peer // {127, 0, 0, 1}, conn) do
```
Comment (josevalim) [inline lib/dynamo/connection/test.ex]: Same here. :)

**Comment by bbrietzke** [comment]
I made the adjustments to allow the the peer value to be set.

Let me know if this is what you had envisioned.

**Comment by devinus** [comment]
:+1:

**Comment by bbrietzke** [comment]
Thank you!

**Pull request merged**